docs/arc42.md

```diff
@@ -0,0 +1,741 @@
+- **Decentralized Design**: Eliminates central authority dependencies.
+
+---
+
+## 3. System Scope and Context
+
+### 3.1 Business Context
+The mediator facilitates routing of DIDComm messages for agents unable to maintain direct communication channels. It acts as an intermediary, ensuring reliability and privacy.
+
+### 3.2 External Components
+- **DIDComm Agents**: Sending and receiving agents.
+- **Transport Protocols**: Various transport layers, e.g., HTTP or WebSockets.
+- **Cloud Infrastructure**: For scalable and fault-tolerant deployments.
+
+### 3.3 Component Diagram
+```mermaid
+graph LR
+    Agent1[Agent 1] --> Transport1
+    Transport1 --> Mediator1
+    Mediator1 --> Mediator2
+    Mediator2 --> Transport2
+    Transport2 --> Agent2[Agent 2]
+```
+
+---
+
+## 4. Solution Strategy
+A microservices approach ensures modularity and scalability. `didcomm-mediator-rs` employs Rust's async capabilities to handle concurrent message routing efficiently.
+
+### Key Features:
+The `didcomm-mediator-rs` project implements several essential protocols in the DIDComm v2 ecosystem, as outlined below:
+
+| **Feature**                             | **Specification Status** | **Implementation Status** |
+|-----------------------------------------|---------------------------|----------------------------|
+| **Mediator Coordination Protocol**      | Adopted                  | ✅ Implemented             |
+| **Pickup Protocol**                     | Adopted                  | ✅ Implemented             |
+| **DID Rotation**                        | Accepted                 | ✅ Implemented             |
+| **Cross-Domain Messaging/Routing Protocol** | Adopted                  | ✅ Implemented             |
+| **Trust Ping Protocol**                 | Adopted                  | ✅ Implemented             |
+| **Discover Features Protocol**          | Adopted                  | ⚪ In Progress              |
+| **Out-of-Band Messaging**               | Adopted                  | ⚪ In Progress              |
+| **Basic Message Protocol**              | Adopted                  | ⚪ In Progress              |
+| **Acknowledgments (Acks)**              | Adopted                  | ❌ Not Implemented         |
+| **Present Proof Protocol**              | Adopted                  | ❌ Not Implemented         |
+
+### Key Highlights:
+1. **Implemented Protocols**:
+   - The mediator is fully equipped with foundational protocols like **Mediator Coordination**, **Pickup**, and **DID Rotation**, ensuring secure and reliable DIDComm message handling.
+   - **Cross-Domain Messaging** facilitates communication across diverse network boundaries, enhancing interoperability.
+
+2. **In Progress Features**:
+   - **Discover Features Protocol** and **Out-of-Band Messaging** are under development to expand functionality.
+   - **Basic Message Protocol** will enable straightforward message exchanges between agents.
+
```

Comment:
i feel like its preferable to remove this highlight here because this will  require frequent modifications, beside the discover features protocol is implemented already

Reply:
also done